### PR TITLE
Change OrderSet add suppress_ immediate_execution parameter.

### PR DESCRIPTION
**Pull request by LGM-Doyle**
This allows orders to be injected into the OrderSet at startup and load
without being executed immediately and then again in MapWnd::InitTurn()

This will allow saved ship designs to be added to its empire by the human client at game start (new game or loaded game).

### CODE REVIEW

File: util/OrderSet.h [hunk 1]
```diff
@@ -52,9 +52,11 @@ class FO_COMMON_API OrderSet {
     //@}
 
     /** \name Mutators */ //@{
-    /** executes a given Order, then stores it in the OrderSet. Returns an index that can be used to reference the
-        order.*/
-    int            IssueOrder(const OrderPtr& order);
+    /** executes \p order unless \p suppress_immediate_execution is true, then
```
Comment (geoffthemedio) [inline util/OrderSet.h]: not clear whether order is stored if `suppress_immediate_execution` is false

File: util/OrderSet.cpp [hunk 2]
```diff
@@ -15,11 +15,15 @@ const OrderPtr OrderSet::ExamineOrder(int order) const {
...
 }
 
-int OrderSet::IssueOrder(const OrderPtr& order) {
+int OrderSet::IssueOrder(const OrderPtr& order, bool suppress_immediate_execution /*= false*/)
+{ return IssueOrder(OrderPtr(order), suppress_immediate_execution); }
+
+int OrderSet::IssueOrder(OrderPtr&& order, bool suppress_immediate_execution /*= false*/) {
     int retval = ((m_orders.rbegin() != m_orders.rend()) ? m_orders.rbegin()->first + 1 : 0);
-    m_orders[retval] = order;
+    auto inserted = m_orders.insert(std::make_pair(retval, std::forward<OrderPtr>(order)));
```
Comment (geoffthemedio) [inline util/OrderSet.cpp]: comment explaining this would be helpful...

File: util/OrderSet.h [hunk 3]
```diff
@@ -52,9 +52,11 @@ class FO_COMMON_API OrderSet {
     //@}
 
     /** \name Mutators */ //@{
-    /** executes a given Order, then stores it in the OrderSet. Returns an index that can be used to reference the
-        order.*/
-    int            IssueOrder(const OrderPtr& order);
+    /** If \p suppress_immediate_execution is true, then execute the \p order on the client.
```
Comment (geoffthemedio) [inline util/OrderSet.h]: do you mean if false then execute?

Comment (LGM-Doyle) [reply util/OrderSet.h]: That is what I meant.  I've fixed it.

**Pull request merged**